cpp/src/utilities/cython.cpp

```diff
@@ -140,155 +259,51 @@ void populate_graph_container(graph_container_t& graph_container,
-    } break;
-    case graphTypeEnum::GraphCSCViewDouble: {
-      graph_ptr_union.GraphCSCViewDoublePtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    case graphTypeEnum::GraphCOOViewDouble: {
-      graph_ptr_union.GraphCOOViewDoublePtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    default: {
-      CUGRAPH_FAIL("unexpected weight type");
-    }
-  }
-}
+
+////////////////////////////////////////////////////////////////////////////////
 
 // Wrapper for calling Louvain using a graph container
 template <typename weight_t>
 std::pair<size_t, weight_t> call_louvain(raft::handle_t const& handle,
                                          graph_container_t const& graph_container,
+                                         void* identifiers,
                                          void* parts,
                                          size_t max_level,
                                          weight_t resolution)
 {
-  std::pair<size_t, float> results;
+  std::pair<size_t, weight_t> results;
 
   // FIXME: the only graph types currently in the container have ints for
   // vertex_t and edge_t types. In the future, additional types for vertices and
   // edges will be available, and when that happens, additional castings will be
   // needed for the 'parts' arg in particular. For now, it is hardcoded to int.
-  if (graph_container.graph_ptr_type == graphTypeEnum::GraphCSRViewFloat) {
+  if (graph_container.graph_ptr_type == graphTypeEnum::graph_t_float_mg) {
+    results = louvain(handle,
+                      graph_container.graph_ptr_union.graph_t_float_mg_ptr->view(),
+                      reinterpret_cast<int*>(parts),
+                      max_level,
+                      static_cast<float>(resolution));
+
+  } else if (graph_container.graph_ptr_type == graphTypeEnum::graph_t_double_mg) {
+    results = louvain(handle,
+                      graph_container.graph_ptr_union.graph_t_double_mg_ptr->view(),
+                      reinterpret_cast<int*>(parts),
+                      max_level,
+                      static_cast<double>(resolution));
+  } else if (graph_container.graph_ptr_type == graphTypeEnum::GraphCSRViewFloat) {
+     //     if (graph_container.graph_ptr_type == graphTypeEnum::GraphCSRViewFloat) {
+    graph_container.graph_ptr_union.GraphCSCViewFloatPtr->get_vertex_identifiers(
+      static_cast<int32_t*>(identifiers));
     results = louvain(handle,
                       *(graph_container.graph_ptr_union.GraphCSRViewFloatPtr),
                       reinterpret_cast<int*>(parts),
                       max_level,
                       static_cast<float>(resolution));
-  } else {
+  } else if (graph_container.graph_ptr_type == graphTypeEnum::GraphCSRViewDouble) {
+    graph_container.graph_ptr_union.GraphCSCViewDoublePtr->get_vertex_identifiers(
+      static_cast<int32_t*>(identifiers));
```

Comment:
What happens if vertex_t is int64_t? Is this path irrelevant or it's not supported?

Reply:
To be addressed in next PR (see comment above).

Reply:
Looked closer at adding support for int64_t and talked with @ChuckHastings and realized we wouldn't support int64_t vertex types for legacy graph classes which are only used for SG.